x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/field_select.tsx

```diff
@@ -168,60 +171,73 @@ export function FieldSelect({
     existingFields,
     markAllFieldsCompatible,
   ]);
+  const comboBoxRef = useRef<HTMLInputElement>(null);
+  const labelWidth =
+    (comboBoxRef.current?.clientWidth || DEFAULT_COMBOBOX_WIDTH) - COMBOBOX_PADDINGS;
```

Comment:
Reading from `.clientWidth` triggers a reflow should be avoided unless necessary. I looked into how EUI does this for the combo box, and what they're doing is:

1. Calculating the initial size
2. Updating the size measurements when the window resizes

I recommend taking the same approach.

https://github.com/elastic/eui/blob/master/src/components/combo_box/combo_box_options_list/combo_box_options_list.tsx#L124